trace/opencensus/trace/tracer/context_tracer.py

```diff
@@ -94,23 +97,23 @@ def end_span(self):
         """
         cur_span = self.current_span()
 
-        self._span_stack.pop()
+        if cur_span is None:
+            logging.warning('No active span, cannot do end_span.')
+            return
+
         cur_span.finish()
+        self.span_context.span_id = cur_span.parent_span.span_id
```

Comment:
Is it valid to manually create a span like `Span(parent_span=None,...)`?  In that case, we would need to check for None here, and set `span_id` to 0 or something.  It seems like we have 3 cases:

1) parent_span is a Span object
2) parent_span is a NullContextManager because we got the span_id from an HTTP header
3) parent_span is None because we manually created a root span.

Reply:
Yes we have these 3 cases, I added check for parent_span in span constructor and documented that user should start and end span using RequestTracer instead of directly using the span class.